File: 2017/Python/J3.py
start = input().split()
end = input().split()
t = int(input())

startX = int(start[0])
startY = int(start[1])
endX = int(end[0])
endY = int(end[1])

d = abs(endX - startX) + abs(endY - startY)

if (t-d)%2==0 and t-d>=0:
    print ("Y")
else:
    print ("N")

File: 2017/Java/J3.java
import java.util.*;
public class Main {
	public static void main(String args[]) {
        Scanner input = new Scanner(System.in);
        int startX = input.nextInt();
        int startY = input.nextInt();
        int endX = input.nextInt();
        int endY = input.nextInt();
        int t = input.nextInt();
        int d = Math.abs(endX - startX) + Math.abs(endY - startY);

        if ((t-d)%2==0 && t-d>=0){
            System.out.println("Y");
        }
        else{
            System.out.println("N");
        }
    }
}
